### ISSUE TITLE
`<Breadcrumbs>` specification

### ISSUE THREAD

**Issue by Heydon** [iTwin/stratakit#907]
Use **Breadcrumbs** to indicate the user’s current location within a hierarchy or sequence.

## Basic example

This example describes a hierarchy of nested folders.

### JSX

```html
<Breadcrumbs>
    <Breadcrumbs.Item href="/">Outer folder</Breadcrumbs.item>
    <Breadcrumbs.Item href="/level-two">Intermediate folder</Breadcrumbs.item>
    <Breadcrumbs.Item href="/current-level">Current folder</Breadcrumbs.item>
</Breadcrumbs>
```

This follows the iTwinUI implementation, using `Breadcrumbs.Item` subcomponents.

### Rendered HTML

By default, the last `Breadcrumbs.Item` receives `aria-current="true"`.

```html
<nav aria-label="Breadcrumbs">
    <ol>
        <li>
            <a href="/">Outer folder</a>
        </li>
        <li>
            <a href="/level-two">Intermediate folder</a>
        </li>
        <li>
            <a href="/current-level" aria-current="true">Current colder</a>
        </li>
    </ol>
</nav>
```

### Differences compared with iTwinUI:

* `Breadcrumbs.Item` always renders as an anchor. Rendering `<button>`s inside a navigation region is something of an anti-pattern. If a new location is rendered with JS, there are certain provisions to make this behavior accessible. These are covered in **Usage**.
* The `aria-current` value is always `true`. This is a generic value and works for all scenarios, reducing configuration needs. There is limited value in specific values like `'location'` and `'page'`.
* iTwinUI's truncation is problematic. The items should be styled to wrap, as part of normal document flow, instead.
* Recommending the use of a dropdown menu in combination with the breadcrumb trail should be omitted, unless there are multiple uses in production. This pattern is unconventional and overly complex.
* Redundant list items are not used as separators. For markup hygiene, this should be achieved using CSS.

```css
li + li::before {
  /* separator styling here */
}
```

## Custom `currentIndex` example

In a linear sequence, such as a set of tasks to complete, the current item may not be the last.

```html
<Breadcrumbs currentIndex={1}>
    <Breadcrumbs.Item href="/">Step 1</Breadcrumbs.item>
    <Breadcrumbs.Item href="/2">Step 2</Breadcrumbs.item>
    <Breadcrumbs.Item href="/3">Step 3</Breadcrumbs.item>
</Breadcrumbs>
```

## Custom label example

```html
<Breadcrumbs label="Your location">
    <Breadcrumbs.Item href="/">Outer folder</Breadcrumbs.item>
    <Breadcrumbs.Item href="/level-two">Intermediate folder</Breadcrumbs.item>
    <Breadcrumbs.Item href="/current-level">Current folder</Breadcrumbs.item>
</Breadcrumbs>
```

The default “Breadcrumbs” label is chosen based on GOV.uk’s implementation (and talking with one of the contributors). They felt “breadcrumb” was a recognizable enough term for AT users and that more technical descriptions like “page hierarchy” would be alienating.

However, depending on the scenario and the nature of the navigable content, another label might be more helpful. Hence the provision of the `label` prop.

### `<Breadcrumbs>`

* `currentIndex` (integer; optional; default: -1): The index of the `currentItem` (zero-based)
* `label` (string; optional; default: `'Breadcrumbs'`): The navigation region’s label.

### `<Breadcrumbs.Item>`

* `href` (string; required): The URL for the breadcrumb link, as standard for anchor (`<a>`).

## Documentation notes

_(Notes for the accompanying usage guidance)_

### JavaScript rendering

If rendering content (subsequent to activating a `Breadcrumbs.Item`) is undertaken with JavaScript, there are likely to be cues missing. The following provisions will bring the experience in line with standard document navigation:

1. Change the `document.title` to reflect the new location. The `document.title` should include the label for the newly current `Breadcrumbs.Item`.
2. Update the URL using `history.pushState`. It should be possible to share the URL directly to the current location.
3. Keyboard focus should be moved to a suitable element. In this case: the newly current `Breadcrumbs.Item`. 

These recommendations will need to be translated into React vernacular...
